tests/unit/Diff/StatementListDifferPatcherTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\Diff;
+
+use Wikibase\DataModel\Services\Diff\StatementListDiffer;
+use Wikibase\DataModel\Services\Diff\StatementListPatcher;
+use Wikibase\DataModel\Snak\PropertyNoValueSnak;
+use Wikibase\DataModel\Snak\SnakList;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+class StatementListDifferPatcherTest extends \PHPUnit_Framework_TestCase {
+
+	public function testStatemetsAreMergedWhenPatched() {
+		$guid = 'some-guid';
+		$statementLatest = new Statement(
+			new PropertyNoValueSnak( 1 ),
+			new SnakList( [ new PropertyNoValueSnak( 3 ) ] ),
+			null,
+			$guid
+		);
+		$statementFrom = new Statement(
+			new PropertyNoValueSnak( 1 ),
+			new SnakList( [ new PropertyNoValueSnak( 1 ) ] ),
+			null,
+			$guid
+		);
+		$statementTo = new Statement(
+			new PropertyNoValueSnak( 1 ),
+			new SnakList( [ new PropertyNoValueSnak( 2 ) ] ),
+			null,
+			$guid
+		);
+		$latestStatementList = new StatementList( [ $statementLatest ] );
+
+		$differ = new StatementListDiffer();
+		$patcher = new StatementListPatcher();
+
+		$diff = $differ->getDiff( new StatementList( [ $statementFrom ] ), new StatementList( [ $statementTo ] ) );
+		$patcher->patchStatementList( $latestStatementList, $diff );
+
+		$statement = $latestStatementList->getFirstStatementWithGuid( $guid );
+		$this->assertEquals( new PropertyNoValueSnak( 3 ), $statement->getQualifiers()[0] );
+		$this->assertEquals( new PropertyNoValueSnak( 2 ), $statement->getQualifiers()[1] );
+		$this->assertEquals( 2, $statement->getQualifiers()->count() );
```

Comment:
you can use assertCount

Reply:
Maybe, but the point of this test code is just to present existence of the bug.